### PR TITLE
feat: Add s3 support to tensorboard logging.

### PR DESCRIPTION
**Pull request by KaleabTessera**
## What?
- Support tensorboard logs to s3 url.
## Why?
- So we use s3 urls for tb logs. 
## How?
- Don't parse s3 urls and add `import tensorflow_io`. 
## Extra
-

### CODE REVIEW

**Review by RuanJohn — APPROVED** [review]
Thanks @KaleabTessera. Looks good and the tb logs are visible on AIchor now. Massive win! :partying_face:

**Review by DriesSmit — DISMISSED** [review]
Thanks @KaleabTessera :raised_hands: Really nice work :slightly_smiling_face:

File: mava/utils/loggers/base.py [hunk 1]
```diff
@@ -173,13 +186,20 @@ def make_logger(
...
+                path = f"{self._directory}/{self._time_stamp}/{subdir}/{ self._label}"
+            else:
+                path = f"{self._directory}/{self._time_stamp}/{self._label}"
+            return path
         else:
-            path = str(self._directory / self._time_stamp / self._label)
+            if subdir:
+                path = str(self._directory / self._time_stamp / subdir / self._label)
+            else:
+                path = str(self._directory / self._time_stamp / self._label)
```
Comment (sash-a) [inline mava/utils/loggers/base.py:199]: not following what's happening here, if it's not a string what is it and what does the divide operator do?

File: mava/utils/loggers/base.py [hunk 2]
```diff
@@ -173,13 +186,20 @@ def make_logger(
         return logger
 
     def _path(self, subdir: Optional[str] = None) -> str:
-        if subdir:
-            path = str(self._directory / self._time_stamp / subdir / self._label)
+        if isinstance(self._directory, str):
+            if subdir:
+                path = f"{self._directory}/{self._time_stamp}/{subdir}/{ self._label}"
+            else:
+                path = f"{self._directory}/{self._time_stamp}/{self._label}"
```
Comment (sash-a) [inline mava/utils/loggers/base.py:193]: Maybe would be cleaner to make a list of this stuff, ignore the empty/None items and then `"/".join(path)`

**Review by sash-a — CHANGES_REQUESTED** [review]
Minor code style stuff but thanks for this major QOL improvement

File: mava/utils/loggers/base.py [hunk 3]
```diff
@@ -61,8 +67,12 @@ def __init__(
         """Initialise logger."""
         self._label = label
 
-        if not isinstance(directory, Path):
-            directory = Path(directory)
+        is_s3_url = isinstance(directory, str) and "s3" in directory
+        # Don't modify path if s3 url, use url string.
+        # Note this requires tensorflow_io to be installed.
+        if not is_s3_url:
+            if not isinstance(directory, Path):
```
[the remark this answers is not in the document]
Comment (KaleabTessera) [reply mava/utils/loggers/base.py]: Done.

File: mava/utils/loggers/base.py (hunk 1 above)
Comment (KaleabTessera) [reply mava/utils/loggers/base.py:198]: So the divider does a similiar thing to the string case. You just can't do `self._directory / ...` with strings (without f strings). Here for a path, you can do this without converting the path object to a string and a couple lines below we do some path cleanup that ensures it is a valid path (`paths.process_path`).

File: mava/utils/loggers/base.py (hunk 2 above)
Comment (KaleabTessera) [reply mava/utils/loggers/base.py:192]: I agree that would be cleaner, but I prefer the readability of how it looks now. Do you think the cleaner implementation is worth the potential readability hit?

Comment (sash-a) [reply mava/utils/loggers/base.py:192]: Ye I thought it might be the same. So can leave as is

File: mava/utils/loggers/base.py (hunk 1 above)
Comment (sash-a) [reply mava/utils/loggers/base.py:198]: Ok nice

**Review by sash-a — APPROVED** [review]
LGTM @KaleabTessera :fire:

**Review by RuanJohn — APPROVED** [review]
Thanks @KaleabTessera :raised_hands:

**Pull request merged**